validphys2/src/validphys/filters.py

```diff
@@ -99,18 +101,63 @@ class FatalRuleError(Exception):
     """Exception raised when a rule application failed at runtime."""
 
 
+@dataclasses.dataclass(frozen=True)
+class FilterDefaults:
+    """
+    Dataclass carrying default values for filters (cuts) taking into
+    account the values of ``q2min``, ``w2min`` and ``maxTau``.
+    """
+    q2min: float
+    w2min: float
+    maxTau: float
+
+    def to_dict(self):
+        return dataclasses.asdict(self)
+
+
+@dataclasses.dataclass(frozen=True)
+class FilterRule:
+    """
+    Dataclass which carries the filter rule information.
+    """
+    dataset: str = None
+    process_type: str = None
+    rule: str = None
+    reason: str = None
+    local_variables: Mapping[str, Union[str, float]] = None
+    PTO: str = None
+    FNS: str = None
+    IC: str = None
+
+    def to_dict(self):
+        rule_dict = dataclasses.asdict(self)
+        filtered_dict = {k: v for k, v in rule_dict.items() if v is not None}
+        return filtered_dict
+
+
+@dataclasses.dataclass(frozen=True)
+class AddedFilterRule(FilterRule):
+    """
+    Dataclass which carries extra filter rule that is added to the
+    default rule.
+    """
+    pass
```

Comment:
as said above, I don't think we should separate AddedFilterRules and FilterRules, they should be at the same level once they are read.